packages/runtime/lifecycle/src/api.ts

```diff
@@ -28,14 +38,177 @@ export type RuntimeLifecycle = {
 export type ContractsDI = WalletDI & RestDI;
 
 export type CreateContractRequest = {
-  contract: Contract;
-  roles?: RolesConfig;
+  /**
+   * A Marlowe Contract or a Merkleized One (referred by its source Id) to create over Cardano
+   * @see Large/Deep Contracts Support (Contract Merkleization) and `@marlowe.io/language-core`
+   */
+  contractOrSourceId: ContractOrSourceId;
+
+  /**
+   * The Marlowe Runtime utilizes this Optional field to set a stake address
+   * where to send staking rewards for the Marlowe script outputs of this contract.
+   */
+  stakeAddress?: StakeAddressBech32;
+  /**
+   * @experimental
+   * Thread Roles are a details of implementation within the runtime. It allows provide a custom name
+   * if the thread role name is conflicting with other role names used.
```

Comment:
I'd change this to "The Thread Roles capability is an implementation details of the runtime. It allows you to provide a custom name if the thread role name is conflicting with other role names used."